solr/core/src/java/org/apache/solr/core/NodeConfig.java

```diff
@@ -211,6 +215,17 @@ private NodeConfig(
     if (null == this.solrHome) throw new NullPointerException("solrHome");
     if (null == this.loader) throw new NullPointerException("loader");
 
+    if (this.clusterPlugins != null
+        && this.clusterPlugins.length > 0
+        && !ClusterPluginsSourceConfigurator.resolveClassName()
+            .equals(NodeConfigClusterPluginsSource.class.getName())) {
+      throw new SolrException(
```

Comment:
Maybe it can be a warning instead of an exception? Not entirely sure, but maybe people will have those in `solr.xml`, change the source later and want to keep them. Or maybe they could be injected into the cluster API as defaults at some point.

Reply:
I don't think we'd be doing users any favors by allowing/encouraging ambiguity in configs, and ignoring some of it based on settings. IMO the exception is correct here.